pennylane/gradients/finite_difference.py

```diff
@@ -22,117 +22,155 @@
+def finite_diff_coeffs(n, approx, strategy):
+    r"""Generate the finite difference shift values and corresponding
+    term coefficients for various derivative order, approximation accuracy,
+    and strategy.
 
     Args:
-        n (int): Positive integer specifying the derivative. ``n=1``
-            corresponds to the first derivative.
-        order (int): Positive integer referring to the accuracy of the
-            returned stencil. E.g., ``order=1`` corresponds to a stencil
-            that returns a first-order approximation to the derivative.
-        form (str): one of ``"forward"``, ``"center"``, or ``"backward"``.
+        n (int): Positive integer specifying the order of the derivative. For example, ``n=1``
+            corresponds to the first derivative, ``n=2`` the second derivative, etc.
+        approx (int): Positive integer referring to the approximation order of the
+            returned coefficients. E.g., ``approx=1`` corresponds to the
+            first-order approximation to the derivative.
+        strategy (str): One of ``"forward"``, ``"center"``, or ``"backward"``.
+            For the ``"forward"`` strategy, the finite-difference shifts occur at the points
+            :math:`x_0, x_0+h, x_0+2h,\dots`, where :math:`h` is some small
+            stepsize. The ``"backwards"`` strategy is similar, but in
+            reverse: :math:`x_0, x_0-h, x_0-2h, \dots`. Finally, the
+            ``"center"`` strategy results in shifts symmetric around the
+            unshifted point: :math:`\dots, x_0-2h, x_0-h, x_0, x_0+h, x_0+2h,\dots`.
 
     Returns:
-        array[float]: a ``(2, N)`` array. The first row corresponds to the
+        array[float]: A ``(2, N)`` array. The first row corresponds to the
         coefficients, and the second row corresponds to the shifts.
 
     **Example**
 
-    >>> finite_diff_stencil(n=1, order=1, form="forward")
+    >>> finite_diff_coeffs(n=1, approx=1, strategy="forward")
     array([[-1.,  1.],
            [ 0.,  1.]])
-    >>> finite_diff_stencil(n=1, order=2, form="center")
+
+    The first row corresponds to the coefficients, and the second corresponds
+    to the shifts. For example, this results in the linear combination:
+
+    .. math:: \frac{-y(x_0) + y(x_0 + h)}{h}
+
+    where :math:`h` is the finite-difference step-size.
+
+    More examples:
+
+    >>> finite_diff_coeffs(n=1, approx=2, strategy="center")
     array([[-0.5,  0.5],
            [-1. ,  1. ]])
-    >>> finite_diff_stencil(n=2, order=2, form="center")
+    >>> finite_diff_coeffs(n=2, approx=2, strategy="center")
     array([[-2.,  1.,  1.],
            [ 0., -1.,  1.]])
     """
     if n < 1 or not isinstance(n, int):
         raise ValueError("Derivative order n must be a positive integer.")
 
-    if order < 1 or not isinstance(order, int):
-        raise ValueError("Accuracy order must be a positive integer.")
+    if approx < 1 or not isinstance(approx, int):

```

Comment:
I kind of liked the original `order` for this term... what about `approx_order`?  (I don't have particularly strong feelings about it though, so feel free to ignore)

Reply:
I liked the original order as well, but I admit it was easy to confuse it with 'nth order derivative'

Reply:
I also prefer `approx_order` but non-blocking!

Reply:
done!